apps/web/components/Embed.tsx

```diff
@@ -726,6 +769,274 @@ const ChooseEmbedTypesDialogContent = () => {
   );
 };
 
+const EmailEmbed = ({ eventType, username }: { eventType?: any; username: string }) => {
+  const { t } = useLocale();
+
+  const [timezone] = useTimePreferences((state) => [state.timezone]);
+
+  const [selectTime, setSelectTime] = useState(false);
+
+  useInitializeBookerStore({
+    username,
+    eventSlug: eventType?.slug ?? "",
+    eventId: eventType?.id,
+    layout: BookerLayouts.MONTH_VIEW,
+  });
+
+  const [selectedDate] = useBookerStore((state) => [state.selectedDate], shallow);
+
+  return (
+    <div className="flex flex-col">
+      <div className="mb-[9px] font-medium">
+        <Collapsible open>
+          <CollapsibleContent>
+            <div className="text-default text-sm">{t("select_date")}</div>
+            <DatePicker eventSlug={eventType?.slug} />
+          </CollapsibleContent>
+        </Collapsible>
+      </div>
+      {selectedDate ? (
+        <div className="mt-[9px] font-medium ">
+          <Collapsible open>
+            <CollapsibleContent>
+              <div
+                className="text-default mb-[9px] flex cursor-pointer items-center justify-between text-sm"
+                onClick={() => setSelectTime((prev) => !prev)}>
+                <p>{t("select_time")}</p>{" "}
+                <>
+                  {!selectedDate || !selectTime ? <ArrowDown className="w-4" /> : <ArrowUp className="w-4" />}
+                </>
+              </div>
+              {selectTime && selectedDate ? <AvailableTimeSlots eventSlug={eventType?.slug} /> : null}
+            </CollapsibleContent>
+          </Collapsible>
+        </div>
+      ) : null}
```

Comment:
The select time toggle only shows when a date has been selected and the toggle was enabled